sdks/python/apache_beam/io/gcp/dicomio.py

```diff
@@ -164,70 +167,109 @@ class DicomSearch(PTransform):
     }
 
   """
-  def __init__(self, credential=None):
+  def __init__(self, buffer_size=8, max_workers=5, credential=None):
     """Initializes DicomSearch.
     Args:
       credential: # type: Google credential object, if it is specified, the
       Http client will use it to create sessions instead of the default.
     """
+    self.buffer_size = buffer_size
+    self.max_workers = max_workers
```

Comment:
and also, it may be good to enable users to override the DicomHttpApiClient here

Reply:
Added, but for some objects reference issues in the unit test it has to be added in an ugly way in the WriteToDicomStore connector...